homeassistant/components/knx/__init__.py

```diff
@@ -229,6 +232,15 @@ async def async_setup(hass, config):
             "https://www.home-assistant.io/blog/2020/09/17/release-115/#breaking-changes"
         )
 
+    # deprecation warning since 2021.3
+    if CONF_KNX_CONFIG in config[DOMAIN]:
+        _LOGGER.warning(
+            "The 'config_file' option will soon be deprecated. Please replace it with Home Assistant config schema "
```

Comment:
It is deprecated (see code above).

```suggestion
            "The 'config_file' option is deprecated. Please replace it with Home Assistant config schema "
```

Please note, that having both a `cv.deprecated` and a custom message, will now cause the user to have duplicate warnings (with different messages).

Is that intentional?

Reply:
I can remove cv.deprecated again if you like. 

Would it be welcome if I changed `cv.deprecated` to accept custom messages in general?

Reply:
> I can remove cv.deprecated again if you like.

I think it's better to choose one or the other. I don't think we should mailbomb a user with duplicate messages from a user experience perspective. However, I don't know if it was intentional or not, hence my question. So 🤷 

> Would it be welcome if I changed `cv.deprecated` to accept custom messages in general?

We didn't have had much need for it in general. As most are just removed after it has been automatically migrated or imported into the UI. KNX is a special case, so let's not add that for now.

Reply:
Well, this could be the first step to migrate knx to config_flow...
Thanks for review!